src/pages/FavoriteVideoDetailsView/FavoriteVideoDetailsView.page.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import { useParams, useLocation } from 'react-router-dom';
+import ListOfRelatedVideos from '../../components/ListOfRelatedVideos';
+import { storage } from '../../utils/storage';
+import { FAVORITES_LIST_STORAGE_KEY } from '../../utils/constants';
+import {
+  FavoritesBtn,
+  Title,
+  Description,
+  Video,
+  VideoDetails,
+  Grid,
+} from './FavoriteVideoDetailsView.styles';
+
+function FavoriteVideoDetailsView() {
+  const [buttonAction, setButtonAction] = useState('Add');
+  const { id } = useParams();
+  const location = useLocation();
+  const { videoTitle, videoDescription, image } = location.state;
+  const url = 'https://www.youtube.com/embed/' + id;
+  const FavoritesList = storage.get(FAVORITES_LIST_STORAGE_KEY);
+  const favoriteVideo = {
+    id: {
+      kind: 'youtube#video',
+      videoId: id,
+    },
+    snippet: {
+      title: videoTitle,
+      description: videoDescription,
+      thumbnails: {
+        high: {
+          url: image,
+        },
+      },
+    },
+  };
+  const firstFavorite = {
+    items: [favoriteVideo],
+  };
+
+  useEffect(() => {
+    if (FavoritesList?.items) {
+      //Add or Remove?
+      let filterFavorites = FavoritesList.items.filter((e) => e.id.videoId === id);
+      if (filterFavorites.length === 1) {
+        setButtonAction('Remove');
+      } else {
+        setButtonAction('Add');
+      }
+    }
+  }, [FavoritesList, id]);
+  function handleClick() {
+    if (buttonAction === 'Add') {
```

Comment:
since you're using `'Add'` and `'Remove'` in many places in this file.  I recommend you use a constant for them to minimize the probability on errors on typos.

```javascript
const Actions = {
   ADD = 'Add',
   REMOVE = 'Remove'
}
```

or

```javascript
const ACTION_ADD = 'Add';
const ACTION_REMOVE = 'Remove';
```